pandas/_libs/window/aggregations.pyx

```diff
@@ -1285,46 +1285,54 @@ def roll_quantile(ndarray[float64_t, cast=True] values, ndarray[int64_t] start,
-                            if idx % 2 == 0:
-                                output[i] = skiplist_get(skiplist, idx, &ret)
+                    for j in range(0, Nq):
+                        quantile = quantiles[j]
+                        idx_with_fraction = quantile * (nobs - 1)
+                        idx = <int>idx_with_fraction
+
+                        if idx_with_fraction == idx:
+                            # no need to interpolate
+                            output[i, j] = skiplist_get(skiplist, idx, &ret)
+                            continue
+
+                        if interpolation_type == LINEAR:
+                            vlow = skiplist_get(skiplist, idx, &ret)
+                            vhigh = skiplist_get(skiplist, idx + 1, &ret)
+                            output[i, j] = ((vlow + (vhigh - vlow) *
+                                            (idx_with_fraction - idx)))
+                        elif interpolation_type == LOWER:
+                            output[i, j] = skiplist_get(skiplist, idx, &ret)
+                        elif interpolation_type == HIGHER:
+                            output[i, j] = skiplist_get(skiplist, idx + 1, &ret)
+                        elif interpolation_type == NEAREST:
+                            # the same behaviour as round()
+                            if idx_with_fraction - idx == 0.5:
+                                if idx % 2 == 0:
+                                    output[i, j] = skiplist_get(skiplist, idx, &ret)
+                                else:
+                                    output[i, j] = skiplist_get(
+                                        skiplist, idx + 1, &ret)
+                            elif idx_with_fraction - idx < 0.5:
+                                output[i, j] = skiplist_get(skiplist, idx, &ret)
                             else:
-                                output[i] = skiplist_get(
-                                    skiplist, idx + 1, &ret)
-                        elif idx_with_fraction - idx < 0.5:
-                            output[i] = skiplist_get(skiplist, idx, &ret)
-                        else:
-                            output[i] = skiplist_get(skiplist, idx + 1, &ret)
-                    elif interpolation_type == MIDPOINT:
-                        vlow = skiplist_get(skiplist, idx, &ret)
-                        vhigh = skiplist_get(skiplist, idx + 1, &ret)
-                        output[i] = <float64_t>(vlow + vhigh) / 2
+                                output[i, j] = skiplist_get(skiplist, idx + 1, &ret)
+                        elif interpolation_type == MIDPOINT:
+                            vlow = skiplist_get(skiplist, idx, &ret)
+                            vhigh = skiplist_get(skiplist, idx + 1, &ret)
+                            output[i, j] = <float64_t>(vlow + vhigh) / 2
             else:
-                output[i] = NaN
+                for j in range(0, Nq):
+                    output[i, j] = NaN
```

Comment:
Any performance difference for large arrays if we just assign an array instead of looping?

`output[i, :] = np.array([np.nan] * Nq)`

Think a similar pattern exists above with `tmp`

Reply:
It is probably easier to just make the change and try it.

pandas/_libs/window/aggregations.pyx:1327:22: Constructing Python slice object not allowed without gil

Error compiling Cython file:
------------------------------------------------------------
...
                        elif interpolation_type == MIDPOINT:
                            vlow = skiplist_get(skiplist, idx, &ret)
                            vhigh = skiplist_get(skiplist, idx + 1, &ret)
                            output[i, j] = <float64_t>(vlow + vhigh) / 2
            else:
                output[i, :] = NaN
                              ^
------------------------------------------------------------

pandas/_libs/window/aggregations.pyx:1327:31: Converting to Python object not allowed without gil